pkg/dsync/drwmutex.go

```diff
@@ -339,15 +339,31 @@ func lock(ctx context.Context, ds *Dsync, locks *[]string, id, source string, is
 	return quorumLocked
 }
 
-// checkQuorumUnlocked determines whether we have unlocked the required quorum of underlying locks or not
-func checkQuorumUnlocked(locks *[]string, quorum int) bool {
-	count := 0
-	for _, uid := range *locks {
-		if uid == "" {
-			count++
+// checkFailedUnlocks determines whether we have sufficiently unlocked all
+// resources to ensure no deadlocks for future callers
+func checkFailedUnlocks(locks *[]string, tolerance int) bool {
```

Comment:
```suggestion
func checkFailedUnlocks(locks []string, tolerance int) bool {
```
If it will crash with a a nil pointer, let's make the caller deference it. We don't have intentions of changing the referenced slice size anyways.